core/quality/morqa_transport.go

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package quality
+
+import (
+	"fmt"
+
+	"github.com/mysteriumnetwork/metrics"
+)
+
+var (
+	emptyMetric = &metrics.Event{}
+)
+
+// NewMORQATransport creates transport allowing to send events to Mysterium Quality Oracle - MORQA
+func NewMORQATransport(morqaClient *MysteriumMORQA) *morqaTransport {
+	return &morqaTransport{
+		morqaClient: morqaClient,
+	}
+}
+
+type morqaTransport struct {
+	morqaClient *MysteriumMORQA
+}
+
+func (transport *morqaTransport) SendEvent(event Event) error {
+	if metric := mapEventToMetric(event); metric != emptyMetric {
+		return transport.morqaClient.SendMetric(metric)
+	}
+
+	return fmt.Errorf("event not implemented")
+}
+
+func mapEventToMetric(event Event) *metrics.Event {
+	switch event.EventName {
+	case startupEventName:
+		return &metrics.Event{
+			IsProvider: false,
```

Comment:
The default value is `false`, we can remove this line.

Reply:
I think for readability is a good think to have it, but for me still question remains - why it is 'false'? Don't we need to send provider startup events?

Reply:
Still general startup now, and we dont event know identity at that time. Solving it in next PR